packages/hash/frontend/src/components/layout/PageSidebar/WorkspaceSwitcher.tsx

```diff
@@ -62,236 +107,81 @@ export const WorkspaceSwitcher: VFC<WorkspaceSwitcherProps> = () => {
           }}
           variant="smallTextLabels"
         >
-          {truncateText(user?.properties.preferredName ?? "User")}
+          {truncateText(activeWorkspace.name)}
```

Comment:
I defer to @MaggieAppleton on this (but @benwerner01 is right re: my general preference not to hyphenate or hard split words across lines where avoidable...caveats abound).

Reply:
If this doesn't get addressed before merging, it'd be good to capture in a follow-up task.

Reply:
Not sure how I missed this. I initially setup text-overflow:ellipsis but decided to go with the js approach because I didn't want to set an explicit width/max-width on the text. Didn't consider the fact that letters have varying widths. I've removed the js approach and just stuck to css in this commit => [74834fe](https://github.com/hashintel/hash/pull/409/commits/74834fe1736692b20c679fdc867cff86dcb18bbf).
Just to confirm, the initial figma designs truncated long text and that's why I went ahead with doing that. I dropped a comment on [figma]( https://www.figma.com/file/gydVGka9FjNEg9E2STwhi2?node-id=182:5746#173187009) for Maggie to decide on what approach we go with. If we are to wrap overflowing text, then I will create a follow-up task

Reply:
Great, thanks for addressing 👍

Reply:
General thoughts on long text: I agree we should wrap overflowing text onto second lines in cases where there's enough space and it doesn't throw the layout off. That said, in cases like this workspace switcher two lines would throw off the layout too much, so we should truncate with the ellipsis approach in the preview. We can then show the full text wrapped onto multiple lines in the dropdown menu.

But another important element to this is user should always be able to see the full text in a tooltip when they hover over the truncated version. For mobile devices where users can't hover the standard approach is to slowly scroll the text in a marquee after they tap, but that only works for things that aren't links or buttons so wouldn't apply here.  

To accommodate extra long workspace names I adjusted the width of the workspace switcher slightly (now 280px) and wrote up some more detailed frames of how it should behave. Figma frames: https://www.figma.com/file/gydVGka9FjNEg9E2STwhi2/HASH-Editor?node-id=2420%3A123220

![image](https://user-images.githubusercontent.com/5599295/161525278-a1c72794-1047-49e1-8e9a-d9612e77670c.png)